### PR TITLE
fix: https://github.com/ctripcorp/apollo/issues/2111

### PR DESCRIPTION
**Pull request by bkblack**
fix the issue : https://github.com/ctripcorp/apollo/issues/2111

### CODE REVIEW

**Comment by nobodyiam** [comment]
Thanks, will take a detailed look soon!

**Comment by bkblack** [comment]
ok，i will update the commit with apollo code style!


 
---Original---

File: apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/controller/NotificationControllerV2.java [hunk 1]
```diff
@@ -131,25 +135,10 @@ public NotificationControllerV2(
...
-        getApolloConfigNotifications(namespaces, clientSideNotifications, watchedKeysMap,
-            latestReleaseMessages);
-
-    if (!CollectionUtils.isEmpty(newNotifications)) {
-      deferredResultWrapper.setResult(newNotifications);
-    } else {
+    //1、set deferredResult before check, for avoid more waiting
+    //If check before set deferredResult and release after check and release before set deferredResult,
+    //it will wait for more time.
+    {
```
Comment (nobodyiam) [inline apollo-configservice/src/main/java/com/ctrip/framework/apollo/configservice/controller/NotificationControllerV2.java]: This empty code block is not necessary, it's also better to remove it.

**Comment by bkblack** [comment]
Sorry! The git commit history has been contaminated. I will resubmit pr.

**Comment by bkblack** [comment]
new pr:  https://github.com/ctripcorp/apollo/pull/2255

**Pull request closed without merging**